In [1]:
import cv2
import matplotlib.pyplot as plt
import sys
import os
import numpy as np
import plotly.graph_objects as go


# Add the 'src' directory to the Python path
sys.path.append(os.path.join(os.getcwd(), 'src'))

In [2]:
# define sand patch coordinates to measure average intensity
x=600
y=900
size=100


In [3]:
def plot_histograms():
    fig = go.Figure()  # Create a figure for the combined plot
    
    for filename in os.listdir('data/images'):
        img = cv2.imread(os.path.join('data/images', filename), cv2.IMREAD_GRAYSCALE)
        hist = cv2.calcHist([img], [0], None, [256], [0, 256]).flatten()  # Flatten to 1D array for Plotly

        # calculate the average pixel value for the whole image
        avg_img = np.mean(img)
        print(f"Average pixel value for {filename}: {avg_img}")
        
        # calculate the average pixel value for the patch
        patch = img[y:y+size, x:x+size]
        avg_patch = np.mean(patch)
        print(f"Average pixel value for the sand patch in {filename}: {avg_patch}")
        
        # Add the histogram as a trace to the figure
        fig.add_trace(go.Scatter(x=np.arange(256), y=hist, mode='lines', name=filename))

    # Update the layout to add title and axis labels
    fig.update_layout(
        title="Histograms of All Images",
        xaxis_title="Pixel Intensity",
        yaxis_title="Frequency",
        showlegend=True
    )
    
    fig.show()  # Show the interactive plot

plot_histograms()

Average pixel value for 0.jpg: 120.22312741126544
Average pixel value for the sand patch in 0.jpg: 118.0861
Average pixel value for 0_empty.jpg: 120.2325390625
Average pixel value for the sand patch in 0_empty.jpg: 118.0917
Average pixel value for 1.jpg: 125.89082272376544
Average pixel value for the sand patch in 1.jpg: 174.7268
Average pixel value for 2.jpg: 121.40615788966049
Average pixel value for the sand patch in 2.jpg: 162.5708
Average pixel value for 3.jpg: 131.75544656635802
Average pixel value for the sand patch in 3.jpg: 182.2547
Average pixel value for 4.jpg: 133.66188368055555
Average pixel value for the sand patch in 4.jpg: 185.7664
Average pixel value for 5.jpg: 130.02435667438272
Average pixel value for the sand patch in 5.jpg: 182.1072
Average pixel value for 6.jpg: 130.9078732638889
Average pixel value for the sand patch in 6.jpg: 178.992
Average pixel value for 7.jpg: 124.01099247685185
Average pixel value for the sand patch in 7.jpg: 169.3087
Average pixel value fo

two different types of images: 
- 
    -  0.jpg

- 
    -  7.jpg
    -  9.jp

-
    -  1.jpg
    -  2.jpg
    -  8.jpg

- Best parameters 
    - noise_reduction:  Median  
    - filter_size:  3  
    - clahe:  False  
    - grid_size:  8  
    - pct:  35.0  
    - static_foreground_removal:  True  
    - dilation:  True  
    - opening:  True  
    - closing:  True  
    - smoothing_after_morph:  False
- one with less sun and more texture in the sand > 130 intensity 
    -  3.jpg
    -  4.jpg
    -  5.jpg
    -  6.jpg
- Best parameters 
    - noise_reduction:  Gaussian  
    - filter_size:  3  
    - clahe:  True  
    - grid_size:  8  
    - pct:  50  
    - static_foreground_removal:  True  
    - dilation:  True  
    - opening:  False  
    - closing:  True  
    - smoothing_after_morph:  True

Threshold could be avg intensity value of 130